vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -167,8 +168,28 @@ def save_kv_layer(self, layer_name: str, kv_layer: torch.Tensor,
         """NixlConnector does not save explicitly."""
         pass
 
-    def wait_for_save(self):
+    def wait_for_save(self, requests : Optional[NewRequestData]):
```

Comment:
This change will need to upstream: https://github.com/vllm-project/vllm/blob/main/vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py#L220

May you explain the design here, for keeping track pf both kv_caches_cpu and kv_caches.